### PR TITLE
Generalize the event store using go generics

### PR DESCRIPTION
**Pull request by Praveenrajmani**
## Description

This PR makes the event store generic so that it can be used by other targets like logger webhook targets, audit webhook targets and audit kafka targets etc.

## Motivation and Context

We will be implementing `queue_dir` support for audit and logger targets soon. For this, we need to generalize the current event store.

## How to test this PR?

Just some sanity tests to make sure the `queue_dir` functionality in event notifications work fine.

- Configure any notification target with `queue_dir` enabled
- Test event store functionality. ie, take down the target, send events and check if the events are properly queued in the provided `queue_dir` and then bring the target up and check if the events are getting replayed properly.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by Praveenrajmani** [comment]
This PR requires https://github.com/minio/pkg/pull/62

File: go.mod [hunk 1]
```diff
@@ -50,7 +50,7 @@ require (
 	github.com/minio/madmin-go/v2 v2.0.17
 	github.com/minio/minio-go/v7 v7.0.50
 	github.com/minio/mux v1.9.0
-	github.com/minio/pkg v1.6.5
+	github.com/minio/pkg v1.6.6-0.20230330040824-5db111e5f63c
```
Comment (Praveenrajmani) [inline go.mod:53]: using `github.com/minio/pkg@main` currently, will change the tag once we make a release of `minio/pkg`

File: internal/store/queuestore.go [hunk 2]
```diff
@@ -15,49 +15,59 @@
...
 	"time"
 
-	"github.com/minio/minio/internal/event"
+	"github.com/google/uuid"
 )
 
 const (
 	defaultLimit = 100000 // Default store limit.
-	eventExt     = ".event"
+	defaultExt   = ".unknown"
```
Comment (klauspost) [inline internal/store/queuestore.go:35]:
```suggestion
	defaultExt   = ".events"
```

**Review by klauspost — APPROVED** [review]
Very nice!

File: internal/store/queuestore.go (hunk 2 above)
Comment (Praveenrajmani) [reply internal/store/queuestore.go:35]: This is an extension used for the items saved in the store. Each and every item that uses the store have its own file extension. Bucket notification events use `.event` already. So, i kept `.unknown` as the default one.

Comment (klauspost) [reply internal/store/queuestore.go:35]: It is .event*s*  - but it shouldn't really be used.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**